### PR TITLE
Dependency freezer script, bumped image deps, and doc-requirements.txt cleanup

### PR DESCRIPTION
**Pull request by consideRatio**
## Script to (re)freeze the image environment

@minrk built a script that I often use like `./dependency freeze --upgrade` to freeze a requirements.in file into a requirements.txt file. It has worked great in Z2JH, with new contributors understanding how to use it right away. I copied that script here.

The Python dependencies relevant to the binderhub Docker image seem to be the requirements of the BinderHub package, and the `google-cloud-logging` package. Due to this I created `requirements.in` with the `google-cloud-logging` package and use the BinderHub's requirements.txt (referenced as binderhub.in) in the root folder of this project to generate /helm-chart/images/binderhub/requirements.txt using an adjusted copy of @minrk's excellent script.

## BinderHub image

- Python 3.7 --> 3.8 in the binderhub image
- Node 12 --> 14 to build the binderhub package HTML/JS resources in the binderhub image build step
- `kubernetes` (Python library) was bumped 0.9.0 --> 0.12.0

## doc-requirements.txt documented and cleaned up

I noted that there was some issues to run `make html`, and that there were some parts that were no longer relevant in doc-requirements.txt, so I trimmed them.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-jupyterhub
 jsonschema
+jupyterhub
+kubernetes
+prometheus_client
 pycurl
+python-json-logger
 tornado>=5.1
-
+traitlets
```
Comment (consideRatio) [inline requirements.txt:12]: This is unchanged except it is sorted and kubernetes is unpinned.

File: helm-chart/images/binderhub/requirements.txt [hunk 2]
```diff
@@ -1,7 +1,55 @@
-pycurl==7.43.0.1
-tornado==6.0.*
-kubernetes==9.0.*
-jupyterhub==1.1.0
-jsonschema==2.6.0
-# Logging sinks to send eventlogging events to
-google-cloud-logging==1.8.*
```
Comment (consideRatio) [inline helm-chart/images/binderhub/requirements.txt:7]: google-cloud-logging is no longer within our image! I need to fix that.

File: helm-chart/images/binderhub/dependencies [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+  chartpress configuration is loaded to ensure the environment is the same as
+  when chartpress builds the tagged image to be published.
+"""
+
+import json
+import os
+from functools import lru_cache
+from subprocess import check_call, check_output
+
+import click
```
Comment (betatim) [inline helm-chart/images/binderhub/dependencies:32]: can you add `click` to `dev-requirements.txt` please

File: helm-chart/images/binderhub/requirements.txt (hunk 2 above)
Comment (betatim) [reply helm-chart/images/binderhub/requirements.txt:7]: I think this is now resolved as the package appears in the output

File: helm-chart/images/binderhub/Dockerfile [hunk 4]
```diff
@@ -38,6 +38,12 @@ RUN pip install --no-cache-dir \
         *.whl \
         -r requirements.txt
 
+# when building the dependencies image
+# add pip-tools necessary for computing dependencies
+# this is not done in production builds by chartpress
```
Comment (betatim) [inline helm-chart/images/binderhub/Dockerfile:43]:
```suggestion
# when building the image used to compute the "frozen"
# dependencies we add `pip-tools` which is for computing
# the dependencies
# It is not added when chartpress builds the image used by
# the helm chart when deploying BinderHub
```

WDYT?

File: doc/conf.py [hunk 5]
```diff
@@ -100,20 +100,18 @@
...
 html_theme_options = {
   "use_edit_page_button": True,
   "github_url": "https://github.com/jupyterhub/binderhub",
   "twitter_url": "https://twitter.com/mybinderteam",
 }
+html_context = {
+    "github_user": "jupyterhub",
+    "github_repo": "binderhub",
+    "github_version": "master",
+    "doc_path":"doc",
```
Comment (betatim) [inline doc/conf.py:113]:
```suggestion
    "doc_path": "doc",
```

File: helm-chart/images/binderhub/dependencies (hunk 3 above)
Comment (consideRatio) [reply helm-chart/images/binderhub/dependencies:32]: Good catch thanks, done!

**Review by betatim — APPROVED** [review]

**Comment by consideRatio** [comment]
@betatim wieeee thanks for reviewing and helping me get this merged! :heart: :tada:

**Pull request merged**